lib/utils/compose.ts

```diff
@@ -94,22 +95,62 @@ export function createProject(
 	};
 }
 
+const getRequestRetryParameters = (): RetryParametersObj => {
+	if (
+		process.env.BALENA_CLI_TEST_TYPE != null &&
+		process.env.BALENA_CLI_TEST_TYPE !== ''
+	) {
+		// We only read the test env vars when in test mode.
+		const { intVar } =
+			require('@balena/env-parsing') as typeof import('@balena/env-parsing');
+		// We use the BALENARCTEST namespace and only parse the env vars while in test mode
+		// since we plan to switch all pinejs clients with the one of the SDK and might not
+		// want to have to support these env vars.
+		return {
+			minDelayMs: intVar('BALENARCTEST_API_RETRY_MIN_DELAY_MS'),
+			maxDelayMs: intVar('BALENARCTEST_API_RETRY_MAX_DELAY_MS'),
+			maxAttempts: intVar('BALENARCTEST_API_RETRY_MAX_ATTEMPTS'),
+		};
+	}
+
+	return {
+		minDelayMs: 1000,
+		maxDelayMs: 60000,
+		maxAttempts: 7,
```

Comment:
Shouldn't we allow for customizing it? e.g. `minDelayMs: intVar('BALENARC_API_RETRY_MIN_DELAY_MS') ?? 1000,`

Reply:
The comment right above is trying to explain that part:
```
		// We use the BALENARCTEST namespace and only parse the env vars while in test mode
		// since we plan to switch all pinejs clients with the one of the SDK and might not
		// want to have to support these env vars.
```
given that we will probably adopt the sdk's retry mechanism, and I'm not sure whether at that point we would want to support these env vars, I though it might be better for now to not expose them. If we change our mind later and decide that we do want them, we can do so easily in a minor.